openquake/risklib/asset.py

```diff
@@ -505,78 +511,79 @@ def __fromh5__(self, dic, attrs):
-        deductibles = ['deductible-%s' % name for name in deductible_d]
-        limits = ['insurance_limit-%s' % name for name in limit_d]
-        retro = ['retrofitted'] if first_asset._retrofitted else []
-        float_fields = loss_types + deductibles + limits + retro
-        int_fields = [(str(name), U16) for name in tagnames]
-        tagi = {str(name): i for i, name in enumerate(tagnames)}
-        asset_dt = numpy.dtype(
-            [('lon', F32), ('lat', F32), ('site_id', U32),
-             ('number', F32), ('area', F32)] + [
-                 (str(name), float) for name in float_fields] + int_fields)
-        num_assets = sum(len(assets) for assets in assets_by_site)
-        assetcol = numpy.zeros(num_assets, asset_dt)
-        asset_ordinal = 0
-        fields = set(asset_dt.fields)
-        for sid, assets_ in enumerate(assets_by_site):
-            for asset in assets_:
-                asset.ordinal = asset_ordinal
-                record = assetcol[asset_ordinal]
-                asset_ordinal += 1
-                for field in fields:
-                    if field == 'number':
-                        value = asset.number
-                    elif field == 'area':
-                        value = asset.area
-                    elif field == 'site_id':
-                        value = sid
-                    elif field == 'lon':
-                        value = asset.location[0]
-                    elif field == 'lat':
-                        value = asset.location[1]
-                    elif field == 'occupants':
-                        value = asset.values[the_occupants]
-                    elif field == 'retrofitted':
-                        value = asset._retrofitted
-                    elif field in tagnames:
-                        value = asset.tagidxs[tagi[field]]
-                    else:
-                        try:
-                            name, lt = field.split('-')
-                        except ValueError:  # no - in field
-                            name, lt = 'value', field
-                        # the line below retrieve one of `deductibles` or
-                        # `insurance_limits` ("s" suffix)
-                        value = getattr(asset, name + 's')[lt]
-                    record[field] = value
-        return assetcol
+
+def build_asset_array(assets_by_site, tagnames=()):
+    """
+    :param assets_by_site: a list of lists of assets
+    :param tagnames: a list of tag names
+    :param time_event: a time event string (or None)
+    :returns: an array `assetcol`
+    """
+    for assets in assets_by_site:
+        if len(assets):
+            first_asset = assets[0]
+            break
+    else:  # no break
+        raise ValueError('There are no assets!')
+    loss_types = []
+    occupancy_periods = []
+    for name in sorted(first_asset.values):
+        if name.startswith('occupants_'):
+            period = name.split('_', 1)[1]
+            if period != 'None':
```

Comment:
Please check if it can be None

Reply:
Yes, the check is still needed for the scenario_risk tests